### ISSUE TITLE
Neovide always attached to the terminal on windows.

### ISSUE THREAD

**Issue by yashamon** [neovide/neovide#2250]
**Describe the bug**

**To Reproduce**
1. Go to terminal on windows start neovide from command line
2. neovide session starts but is attached to the terminal
3. Quit terminal and neovide quits.
4. This happens even running neovide via explorer, that is run 
"explorer path/to/neovide"

**Expected behavior**
Neovide does not quit, when quitting the terminal. 

**Screenshots**

**Desktop (please complete the following information):**
 - Windows 11
 - Neovide Version 0.12.1
 - Neovim Version release

This is a new issue, neovide was always detached in the past.

**Comment by fredizzimo** [comment]
You are probably running a debug build, which is actually a console application to make it easier to use stdout debugging. The release build on the other hand should never attach to the terminal. The `--fork/--no--fork` option is ignored on Windows.

The reason for this is that it's almost impossible to make a hybrid console/windows app without using two executables.

**Comment by yashamon** [comment]
How would I check if its a debug build? I think I have the standard
executable from the scoop installation.

**Comment by fredizzimo** [comment]
Hm.. If it's from scoop, then it should be release, and the exe should be the same as the one on our releases here on Github.

Unless that one is wrong as well, I will have someone to check, I can't do it myself right now.

**Comment by Mentalist** [comment]
This is a known issue tracked here: https://github.com/ScoopInstaller/Scoop/issues/1606

**Comment by yashamon** [comment]
I don't think its exact same issue, because scoop neovide didn't use to
attach until about a couple of weeks ago.
https://github.com/ScoopInstaller/Scoop/issues/1606 dates years.

**Comment by fredizzimo** [comment]
Is it possible that you have changed the shim to something else? The default one (https://github.com/kiennq/scoop-better-shimexe) should autodetect gui apps, but others might not.

When the issue was opened, that was not the default.

**Comment by fredizzimo** [comment]
Ah, thats also old information, there was this PR a couple of months ago
* https://github.com/ScoopInstaller/Scoop/pull/5559

So it's possible that it or something else has changed the behaviour.

**Comment by Kethku** [comment]
I just installed scoop on my machine, added the extras bucket, and installed neovide and could not reproduce this issue. I also looked at the installed exe and it appears to be the release build (though I'm not sure exactly how i would go about checking for sure. In anycase the size looked right)

**Comment by yashamon** [comment]
Kethku are you running release version of Win 11 or later?

I will try reinstalling scoop.

It is really weird because running scoop zoom from command line gives a detached zoom session. So I don't think its my scoop executable.

**Comment by yashamon** [comment]
Actually I don't understand what reinstalling means.  I am in C:\Users\username\scoop\apps\scoop\current
git status shows I am on master and working tree clean. Git pull shows everything up to date. 

Do I ignore this, purge the directory and reinstall? That seems totally paranoid.

**Comment by yashamon** [comment]
I reinstalled scoop, no change in behavior. The neovide executable behaves correctly if run directly from the apps directory. So it is a problem with the shimmification algorithm. 

At the moment I am using the following work around. Remove the shims from my Path. Add all the scoop/apps paths to the Path (you can automate this). This is probably not totally fail proof as some apps use paths likes scoop/apps/AppName/current/bin
so you also have to add the bin endings.

**Comment by sid-6581** [comment]
For my scoop apps I also bypass the shims for this reason. I never found the shims (even when changed) to behave properly.

**Comment by yashamon** [comment]
I think one can also just use the shim file to generate a symlink to prefered directory. I can't imagine why that is not the default behavior.

Anyway the issue can probably be closed since this appears to be a scoop issue.

**Comment by fredizzimo** [comment]
Yes, I will close the issue. It would probably be worth opening a follow up issue in the scoop repository, or perhaps the repository for the shim.

**Comment by fredizzimo** [comment]
Did someone try the workaround in
* https://github.com/ScoopInstaller/Scoop/issues/1606

and change the shim with?
```
scoop config shim kiennq
scoop reset *
```

**Comment by fredizzimo** [comment]
On the other hand default scoop shim should also work, if you have this PR
* https://github.com/ScoopInstaller/Scoop/pull/5559

You might have to update scoop and re-install neovide though
```
scoop update
scoop uninstall neovide
scoop install neovide
```

**Comment by SpacewaIker** [comment]
How can I get the above PR for my scoop installation? I tried switching branches in `scoop/apps/scoop/current` but that doesn't work, and I can't find any info about this in the scoop documentation

**Comment by SpacewaIker** [comment]
If I'm not mistaken, it's merged with the develop branch, which is 50 commits ahead of master. I tried the commands you gave, but it didn't fix the issue

**Comment by fredizzimo** [comment]
Sorry for my bad suggestions, I don't have a computer to test things on currently, and I am only going on what's been reported in various other issues.

**Comment by SpacewaIker** [comment]
No worries, I perfectly understand! As an update, however, I found that you can use `scoop config scoop_branch develop` to switch to the develop branch. So I did that, updated scoop, and re-installed neovide. Unfortunately, it doesn't seem to fix the issue... I'm not sure if I'm missing a step to make it work?

**Comment by SpacewaIker** [comment]
I do get `Making C:\Users\username\scoop\shims\neovide.exe a GUI binary.`. I tested with nvim-qt (after reinstalling it) and it also doesn't work... I'm wondering if it might be an issue with how scoop works with my terminal emulator (wezterm) or my shell (nushell)? But in any case it's not related to neovide it seems, so I should probably discuss this on the scoop repo!

**Comment by fredizzimo** [comment]
Yes, I think it's better to continue the discussion there, especially since it's not limited to neovide only and actually detected properly.

**Comment by Manuzor** [comment]
I just upgraded to 0.12.2 and noticed the same issue that neovide seems to stay attached to the terminal. Then I found this issue and did some testing.

First off, I don't use scoop, I build neovide myself using `cargo build --release`. Doing `file neovide.exe` outputs:

```
target\release\neovide.exe: PE32+ executable (GUI) x86-64, for MS Windows, 6 sections
```

So it's clearly the GUI version. What I found out is that it matters what shell I start neovide from. When I run neovide from Nushell or elvish, neovide stays attached. When I run neovide from cmd.exe or pwsh.exe, neovide does not stay attached. I tried it from Windows Terminal and Wezterm but that didn't seem to matter.

It even works when I'm in a Nushell session, and then rum `cmd`, which puts me into an interactive session of `cmd`, and then I run neovide, which does not stay attached. So, in other words:

```sh
$ nu
$$ neovide    # <-- blocks further input
$$ cmd
$$$ neovide   # <-- unblocked
$$$ nu
$$$$ neovide  # <-- blocking again
$$$$ exit
$$$ exit
$$ exit
$ exit
```

**Comment by fredizzimo** [comment]
@Manuzor, can you try some other GUI application, like `notepad.exe` ? 

And also Visual Studio code, which uses a wrapper, that does roughly the following `neovide.exe &2>/dev/null </dev/null`. See this for more information:
* https://github.com/neovide/neovide/issues/2036

**Comment by sid-6581** [comment]
I'm seeing the same thing when launched from MSYS2. Neovide stays attached, notepad and VS Code do not. When launching from `pwsh` or `cmd`, no applications stay attached. As for nvim-qt, it doesn't detach in any shell.

I can look into this and see if there's a solution. I feel like it should work at least in MSYS2 and nushell running natively in Windows.

**Comment by Manuzor** [comment]
> @Manuzor, can you try some other GUI application, like `notepad.exe` ? [...]
Didn't occur to me to try something else. 😅 But yea, notepad stays attached in nushell and elvish, just like neovide.

As for vscode, when I run the `code.cmd` wrapper, it detaches, but it does that only after a second or so while vscode proper is already visible. I assume the wrapper stays attached and then detaches when it kills itself. Looks like this from what I saw in procexp64 as well.

**Comment by Manuzor** [comment]
Another note: My previous version that I built myself was 0.11.2 (according to the file properties) and this version does not stay attached. Looking at procexp64, 0.11.2 ends up as a process on the top level, i.e. there's no parent process.

On the other hand, 0.12.2 ends up as a child process/node of the shell I launched it from. This is the case for nushell as well as cmd.

![image](https://github.com/neovide/neovide/assets/5539930/cd103239-976f-4f98-a06d-a522ac477fa6)

**Comment by fredizzimo** [comment]
One thing worth mentioning, is that the debug mode Neovide is supposed to behave differently and stay attached since it's then a console application. @Kethku, wanted it that way for easier debugging using print statements and similar. 

But in the release mode, we have to create a gui application, or there would be console window popping up every time you launch it.

**Comment by yashamon** [comment]
For me in nushell neovide is only weakly attached. Meaning the shell is
blocked on startup of neovide but killing the terminal does not kill
neovide.

Most gui apps i have totally detach from nushell on startup. So shell is
not blocked. The main exception is xournalpp which is also weakly attached.

It would be good to have full detachment.

**Comment by mabeauchemin** [comment]
> I'm seeing the same thing when launched from MSYS2. Neovide stays attached, notepad and VS Code do not. When launching from `pwsh` or `cmd`, no applications stay attached. As for nvim-qt, it doesn't detach in any shell.
> 
> I can look into this and see if there's a solution. I feel like it should work at least in MSYS2 and nushell running natively in Windows.

It's a work around but running 'start neovide' from MSYS2 detaches.

**Comment by yashamon** [comment]
> > I'm seeing the same thing when launched from MSYS2. Neovide stays attached, notepad and VS Code do not. When launching from `pwsh` or `cmd`, no applications stay attached. As for nvim-qt, it doesn't detach in any shell.
> > I can look into this and see if there's a solution. I feel like it should work at least in MSYS2 and nushell running natively in Windows.
> 
> It's a work around but running 'start neovide' from MSYS2 detaches.

I think you should reopen this as a new bug. It is an important problem. And possibly is why scoop is having issues creating a correct executable.
